src/file_buffer.rs

```diff
@@ -135,7 +135,7 @@ impl<F: Write + Read + Seek> BufFile<F> {
     }
 
     fn fetch_slab(&mut self, mut loc: u64) -> Result<&mut Slab, Error> {
-        loc = loc & !SLAB_MASK;
+        loc = loc & SLAB_MASK;
```

Comment:
You're right - I'm not exactly sure why I did that 🤔 

On my local copy of the code I think I was using loc & SLAB_MASK, which was slowing it down a lot, not what happened though.